bpf/flows_sock.c

```diff
@@ -217,6 +212,22 @@ int socket__http_filter(struct __sk_buff *skb) {
             .flags = flags, 
         };
 
+        // Calculate direction based on first flag received
+        // SYN and ACK mean someone else initiated the connection and this is the INGRESS direction
+        if((flags & (SYN_FLAG | ACK_FLAG)) == (SYN_FLAG | ACK_FLAG)) {
```

Comment:
Tested endpoint `curl http://portquiz.net:60000`
before changes (wrong direction, because `src_port` is lower than `dst_port`)
```
network_flow: transport=6 beyla.ip=192.168.5.15 iface=unknown direction=1 src.address=35.180.139.74 dst.address=192.168.5.15 src.name=35.180.139.74 dst.name=192.168.5.15 src.port=60000 dst.port=44364
network_flow: transport=6 beyla.ip=192.168.5.15 iface=unknown direction=0 src.address=192.168.5.15 dst.address=35.180.139.74 src.name=192.168.5.15 dst.name=35.180.139.74 src.port=44364 dst.port=60000
```

after changes
```
network_flow: transport=6 beyla.ip=192.168.5.15 iface=unknown direction=0 src.address=35.180.139.74 dst.address=192.168.5.15 src.name=35.180.139.74 dst.name=192.168.5.15 src.port=60000 dst.port=44364
network_flow: transport=6 beyla.ip=192.168.5.15 iface=unknown direction=1 src.address=192.168.5.15 dst.address=35.180.139.74 src.name=192.168.5.15 dst.name=35.180.139.74 src.port=44364 dst.port=60000
```